base/loggers/OwnSplitChannel.cpp

```diff
@@ -26,16 +27,54 @@ void OwnSplitChannel::log(const Poco::Message & msg)
         auto matches = masker->wipeSensitiveData(message_text);
         if (matches > 0)
         {
-            logSplit({msg, message_text}); // we will continue with the copy of original message with text modified
+            tryLogSplit({msg, message_text}); // we will continue with the copy of original message with text modified
             return;
         }
 
     }
 
-    logSplit(msg);
+    tryLogSplit(msg);
 }
 
 
+void OwnSplitChannel::tryLogSplit(const Poco::Message & msg)
+{
+    try
+    {
+        logSplit(msg);
+    }
+    catch (...)
+    {
+        MemoryTracker::LockExceptionInThread lock_memory_tracker(VariableContext::Global);
+
+        /// It is better to catch the errors here in order to avoid
+        /// breaking some functionality because of unexpected "File not
+        /// found" (or similar) error.
+        ///
+        /// For example StorageDistributedDirectoryMonitor will mark batch
+        /// as broken, some MergeTree code can also be affected.
+        ///
+        /// Also note, that we cannot log the exception here, since this
+        /// will lead to recursion, using regular tryLogCurrentException().
+        /// but let's log it into the stderr at least.
+        std::string message = "Cannot add message to the log:\n";
+        message += msg.getText();
+        message += "\n";
+        message += getCurrentExceptionMessage(true);
+
+        try
+        {
+            WriteBufferFromFileDescriptor out(STDERR_FILENO, 4096);
```

Comment:
We can also get rid of temporary allocation. We can write std::string contents directly.
Maybe write a loop with `write` and handle EINTR?

Reply:
Yes, this looks better, done.